### ISSUE TITLE
Cannot suppress UnusedVariable

### ISSUE THREAD

**Issue by still-dreaming-1** [vimeo/psalm#5537]
https://psalm.dev/r/25a700869d

**Comment by orklah** [comment]
Please check
https://github.com/vimeo/psalm/issues/5516
https://github.com/vimeo/psalm/issues/5507
https://github.com/vimeo/psalm/issues/5493

**Comment by still-dreaming-1** [comment]
> Please check
> #5516
> #5507
> #5493

Oops. I know I searched for this before posting here, I guess I accidentally only searched open issues. I actually did go through and change all truly unused variables to `$_` before finally trying to just suppress the remaining ones. There is some technical debt in our code we are not ready to deal with that is preventing Psalm from being able to realize the variables are actually used. I think being able to name unused variables as `$_` is a great feature, but is different from suppressing it. In the code I'm working on, renaming the variable would break the code since the variables are used. I don't see any reason you should not be able to suppress it like any other issue with an inline suppression annotation. I was able to suppress into the baseline though, so that is workaround I am using.

**Comment by jpresutti** [comment]
I am seriously asking you to support psalm-suppress for this. Expecting us to change our variable names is poor form. Often, coding standards prevent that.

In addition I have open issues right now where it's a false positive. Being able to suppress it until the issues are fixed would make my life a LOT easier.

**Comment by mj4444ru** [comment]
```php
<?php

$viewName = 'logs';

require __DIR__ . '/view.php';
```

My example code. I can't suppress the problem.

It would be correct to ignore unused variables when including external scripts.

And it would be even better to tell psalm which variables we plan to transfer to the internal code of the included script.
